src/RecipeEditPage.js

```diff
@@ -0,0 +1,57 @@
+import { isEqual } from 'lodash'
+import { useState } from 'react'
+import { Link, useParams } from 'react-router-dom'
+import useSWR from 'swr'
+
+import fetcher, { update } from './lib/fetcher'
+import RecipeForm from './RecipeForm'
+
+export default function RecipeEditPage() {
+  const { id } = useParams()
+  const { data, error, isLoading, isValidating, mutate } = useSWR(
+    `/api/recipes/${id}/`, fetcher
+  )
+  const [firstLoadData, setFirstLoadData] = useState(undefined)
+  const submit = async newData => {
+    const responseData = await mutate(async () => {
+      return (await update(`/api/recipes/${id}/`, newData))
+    })
+    setFirstLoadData(responseData)
+  }
+  let content
+  if (isLoading) {
+    content = <p>Loading current recipe values...</p>
+  } else if (error) {
+    content = <p>Error loading recipe. {error.toString()}.</p>
+  } else if (data.detail) {
+    content = <p>Error loading recipe. {data.detail}.</p>
+  } else {
+    if (firstLoadData === undefined) {
+      setFirstLoadData(data)
+    }
+    let sinceWarning
```

Comment:
Yeh, I figured if SWR was going to be refreshing the data anyway, instead of blocking it I could use it for something good, felt more in line with the idea of what SWR is trying to do.